### PR TITLE
Update NGSI support to use ES6 syntax.

### PR DESCRIPTION
**Pull request by jason-fox**
Similar to https://github.com/telefonicaid/iotagent-node-lib/pull/841#issuecomment-584621453

This   is a **limited** PR updates the files within  `/services/devices/`, `services/ngsi/` and `services\northBound` to properly use `let` and `const`. When processed it would reduce the scope of #831

It only contains automated code updates at the moment. The PR could also be amended to correctly rename the `northBound` subdirectory to `northPort` see: #645

The assumption here is that since none of the other outstanding PRs are amending the NGSI interactions, this would be an opportunity to modernize at least part of the code base.

### CODE REVIEW

**Comment by fgalan** [comment]
Once PR #848 has been merged, I have change the base of this PR to feature/842_ngsi_ld

Please upgrade jason-fox:feature-ngsi-ld-es6 with telefonicaid:feature/842_ngsi_ld so this PR gets ready to merge (note a couple of files are conflicting).

**Comment by jason-fox** [comment]
Merged, updated and removed `console.error()` debug - see https://github.com/telefonicaid/iotagent-node-lib/pull/847#discussion_r387775415

File: test/tools/utils.js [hunk 1]
```diff
@@ -37,11 +35,6 @@ function readExampleFile(name, raw) {
         console.error(JSON.stringify(e));
```
Comment (fgalan) [inline test/tools/utils.js]: Should this one also be removed? Or is pre-existing to your work on NGSI-LD?

(Maybe a grep for `console.` in all the code base could help to find additional cases)

Comment (jason-fox) [reply test/tools/utils.js]: Nope - that is a deliberate console output - see the previous line. It was part of the output of #832

**Pull request merged**